File: StackPushPopOrder/StackPushPopOrder.py
# -*- coding:utf-8 -*-
class Solution:
    def IsPopOrder(self, pushV, popV):
        # write code here
        n = len(pushV)
        m = len(popV)
        if n!=m:
            return False
        s = []
        index = 0
        for i in range(n):
            s.append(pushV[i])
            while len(s)>0 and s[-1]==popV[index]:
                s.pop()
                index+=1
        if len(s)==0:
            return True
        return False

File: StackPushPopOrder/StackPushPopOrder.cpp
class Solution {
public:
    bool IsPopOrder(vector<int> pushV,vector<int> popV) {
        int n = pushV.size();
        int m = popV.size();
        if (n!=m) return false;
        stack<int> s;
        int index = 0;
        for (int i=0;i<n;i++)
        {
            s.push(pushV[i]);
            while(!s.empty() && s.top()==popV[index])
            {
                s.pop();
                index++;
            }
        }
        return s.empty();
    }
};
